controllers/comments.controller.js

```diff
@@ -18,4 +18,23 @@ const getCommentsByArticleId = (req, res, next) => {
     .catch(next)
 }
 
-module.exports = { getCommentsByArticleId }
\ No newline at end of file
+const postCommentByArticleId = (req, res, next) => {
+  const { article_id } = req.params;
+  const { username, body } = req.body;
+
+  // Error handling for invalid data
+  if (isNaN(article_id)) {
+    return res.status(400).send({ msg: "Invalid article ID" })
+  }
+  if (!username || !body) {
+    return res.status(400).send({ msg: "Username and body are required" });
+  }
+
```

Comment:
The above checks whilst not strictly wrong are not necessary - you can let psql deal with them and throw an error itself. If the data you are trying to insert into the database does not match the seed it will reject. You can catch the error in your controller!